com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/plugin/PluginScanner.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020-2022 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+// 
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+// 
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package com.dynamo.bob.plugin;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.lang.reflect.Modifier;
+
+import com.dynamo.bob.Bob;
+import com.dynamo.bob.Project;
+import com.dynamo.bob.IClassScanner;
+import com.dynamo.bob.CompileExceptionError;
+
+
+public class PluginScanner {
+
+	/**
+	 * Find and create instances of classes extending a specific base class
+	 * and located within a specific package. The class must:
+	 * 
+	 * - Not be private
+	 * - Not be abstract
+	 * - Extend the base class
+	 * 
+	 * @param packageName 
+	 * @param pluginBaseClass
+	 * @return List of instances
+	 */
+	public static <T> List<T> createPlugins(String packageName, Class<T> pluginBaseClass) throws CompileExceptionError {
```

Comment:
I think the creation of multiple instances is a bit confusing.
We currently run both compilers, and have no control of the order.

Reply:
Yep. Changed to a single one.